src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3537,18 +3538,26 @@ private BoundCatchBlock BindCatchBlock(CatchClauseSyntax node, ArrayBuilder<Boun
             return new BoundCatchBlock(node, locals, exceptionSource, type, boundFilter, block, hasError);
         }
 
-        private BoundExpression BindCatchFilter(CatchFilterClauseSyntax filter, DiagnosticBag diagnostics)
+        private BoundExpression BindCatchFilter(CatchFilterClauseSyntax filter, bool hasSingleCatch, DiagnosticBag diagnostics)
         {
             BoundExpression boundFilter = this.BindBooleanExpression(filter.FilterExpression, diagnostics);
             if (boundFilter.ConstantValue != ConstantValue.NotAvailable)
             {
-                Error(diagnostics, ErrorCode.WRN_FilterIsConstant, filter.FilterExpression);
+                // Depending on whether the filter constant is true or false, and whether there are other catch clauses,
+                // we suggest different actions
+                var errorCode = boundFilter.ConstantValue.BooleanValue
+                    ? ErrorCode.WRN_FilterIsConstantTrue
+                    : hasSingleCatch
+                        ? ErrorCode.WRN_FilterIsConstantRedundantTryCatch
+                        : ErrorCode.WRN_FilterIsConstantFalse;
+
+                // Since the expression is a constant, the name can be retrieved from the first token
+                Error(diagnostics, errorCode, filter.FilterExpression, boundFilter.Syntax.GetFirstToken().Text);                
```

Comment:
That is a good point... It might be best to use the constant value to say `...filter is constant 'false'...` and `...filter is constant 'true'...` instead of specifying the name (e.g. if `default` is used, then it may be more natural to report `...filter is constant 'false'...` rather than `...filter is constant 'default'...` since the compiler already knows what it evaluates to. Similarly for `(1+1)==2` or `(1+1)!=2`, it would then report as `true` and `false` respectively. 
No need to pass around the name either in that case.